Sources/X509/Extension Types/SubjectKeyIdentifier.swift

```diff
@@ -78,3 +80,17 @@ extension SubjectKeyIdentifier: CertificateExtensionConvertible {
         return try .init(self, critical: false)
     }
 }
+
+extension SubjectKeyIdentifier {
+    /// Construct a ``SubjectKeyIdentifier`` by hashing the given `publicKey` with SHA-1 according to RFC 5280 Section 4.2.1.2.
+    /// - Parameter publicKey: the public key which will be hashed
+    @inlinable
+    public init(hash publicKey: Certificate.PublicKey) {
```

Comment:
I wonder if it's worth adding a shortcut, similar to BoringSSL, where you can just specify `"hash"` and it will use public key from the certificate itself, wdyt?

Reply:
I think I'm broadly opposed to it. It's really quite hard to construct an API where that functions sensibly.